### PR TITLE
Rename create data object task to modify-data-object

### PR DESCRIPTION
**Pull request by ksimon1**
**What this PR does / why we need it**:
Rename create data object task to modify-data-object
After adding the delete option to create-data-object, the name is not accurate.
/hold merge after https://github.com/kubevirt/kubevirt-tekton-tasks/pull/176

**Release note**:
```
Rename create data object task to modify-data-object
```

### CODE REVIEW

File: modules/modify-data-object/pkg/dataobject/operations.go [hunk 1]
```diff
@@ -1,8 +1,8 @@
 package dataobject
 
 import (
-	"github.com/kubevirt/kubevirt-tekton-tasks/modules/create-data-object/pkg/constants"
-	"k8s.io/api/core/v1"
+	"github.com/kubevirt/kubevirt-tekton-tasks/modules/modify-data-object/pkg/constants"
+	v1 "k8s.io/api/core/v1"
```
Comment (0xFelix) [inline modules/modify-data-object/pkg/dataobject/operations.go:5]: Does explicit naming of the import have benefits?

**Review by 0xFelix — APPROVED** [review]
One little nit, otherwise looks good!

File: modules/modify-data-object/vendor/github.com/kubevirt/kubevirt-tekton-tasks/modules/sharedtest/testobjects/datasource/datasource.go [hunk 2]
```diff
@@ -61,6 +61,11 @@ func (d *TestDataSource) WithSourcePVC(name, namespace string) *TestDataSource {
 	return d
 }
 
+func (d *TestDataSource) WithGenerateName(generateName string) *TestDataSource {
```
Comment (0xFelix) [inline modules/modify-data-object/vendor/github.com/kubevirt/kubevirt-tekton-tasks/modules/sharedtest/testobjects/datasource/datasource.go:64]: New helper function in rename PR?

File: modules/modify-data-object/vendor/github.com/kubevirt/kubevirt-tekton-tasks/modules/sharedtest/testobjects/datavolume/datavolumes.go [hunk 3]
```diff
@@ -103,6 +103,11 @@ func (d *TestDataVolume) WithSize(size int64, scale resource.Scale) *TestDataVol
 	return d
 }
 
+func (d *TestDataVolume) WithGenerateName(generateName string) *TestDataVolume {
```
Comment (0xFelix) [inline modules/modify-data-object/vendor/github.com/kubevirt/kubevirt-tekton-tasks/modules/sharedtest/testobjects/datavolume/datavolumes.go:106]: New helper function in rename PR?

**Review by 0xFelix — COMMENTED** [review]
Can you please also make sure to rename the task in https://github.com/kubevirt/tekton-tasks-operator/tree/main/data/tekton-pipelines?

File: modules/modify-data-object/vendor/github.com/kubevirt/kubevirt-tekton-tasks/modules/sharedtest/testobjects/datasource/datasource.go (hunk 2 above)
Comment (ksimon1) [reply modules/modify-data-object/vendor/github.com/kubevirt/kubevirt-tekton-tasks/modules/sharedtest/testobjects/datasource/datasource.go:64]: this is a leftover from previous PR, when someone forget to run make vendor.

File: modules/modify-data-object/vendor/github.com/kubevirt/kubevirt-tekton-tasks/modules/sharedtest/testobjects/datavolume/datavolumes.go (hunk 3 above)
Comment (ksimon1) [reply modules/modify-data-object/vendor/github.com/kubevirt/kubevirt-tekton-tasks/modules/sharedtest/testobjects/datavolume/datavolumes.go:106]: the same

File: modules/modify-data-object/build/modify-data-object/Dockerfile [hunk 4]
```diff
@@ -1,4 +1,19 @@
 FROM registry.access.redhat.com/ubi8/ubi-minimal AS builder
-RUN microdnf install -y golang-1.17.* && microdnf clean all
+RUN microdnf install -y tar gzip && microdnf clean all
+ENV TASK_NAME=modify-data-object \
+    GOFLAGS="-mod=vendor"
+WORKDIR /src/${TASK_NAME}
+RUN curl -L https://go.dev/dl/go1.18.3.linux-amd64.tar.gz | tar -C /usr/local -xzf -
```
Comment (0xFelix) [inline modules/modify-data-object/build/modify-data-object/Dockerfile:6]: Upgrade to go1.18.4? Can we instead use ubi9?

**Review by 0xFelix — APPROVED** [review]
One comment not directly related to this change. Otherwise LGTM!

/lgtm

File: modules/modify-data-object/build/modify-data-object/Dockerfile (hunk 4 above)
Comment (ksimon1) [reply modules/modify-data-object/build/modify-data-object/Dockerfile:6]: we definitely can. I can do it in follow-up PRs

**Pull request merged**